src/controllers/auth.controller.js

```diff
@@ -0,0 +1,182 @@
+  }
+}
+
+async function register(req, res, next) {
+  const { email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.BadRequest('Validation error', errors);
+  }
+
+  await userService.register({ email, password });
+
+  res.send({ message: 'OK' });
+}
+
+async function activate(req, res, next) {
+  const { activationToken } = req.params;
+
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  user.activationToken = null;
+
+  await user.save();
+
+  await sendAuthentication(res, user);
+}
+
+async function login(req, res, next) {
+  const { email, password } = req.body;
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.BadRequest('User with this email does not exist');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.BadRequest('Password is wrong');
+  }
+
+  await sendAuthentication(res, user);
+}
+
+async function refresh(req, res, next) {
+  const { refreshToken } = req.cookies;
+  const userData = jwtService.validateRefreshToken(refreshToken);
+
+  if (!userData) {
+    throw ApiError.Unauthorized();
+  }
+
+  const token = tokenService.getByToken(refreshToken);
+
+  if (!token) {
+    throw ApiError.Unauthorized();
+  }
+
+  const user = await userService.getByEmail(userData.email);
+
+  await sendAuthentication(res, user);
+}
+
+async function logout(req, res, next) {
+  const { refreshToken } = req.cookies;
+  const userData = jwtService.validateRefreshToken(refreshToken);
+
+  res.clearCookie('refreshToken');
+
+  if (userData) {
+    await tokenService.remove(userData.id);
+  }
+
+  res.sendStatus(204);
+}
+
+async function sendAuthentication(res, user) {
+  const userData = userService.normalize(user);
+  const accessToken = jwtService.generateAccessToken(userData);
+  const refreshToken = jwtService.generateRefreshToken(userData);
+
+  await tokenService.save(user.id, refreshToken);
+
+  res.cookie('refreshToken', refreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    httpOnly: true,
+    sameSite: 'none',
+    secure: true,
+  });
+
+  res.send({
+    user: userData,
+    accessToken,
+  });
+}
+
+async function reset(req, res, next) {
+  const { email } = req.body;
+  const user = await userService.getByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('User not found');
+  }
+
+  await userService.resetEmail(email);
+
+  res.send('Check your email for reset');
+}
+
+async function resetPassword(req, res, next) {
+  const { resetToken } = req.params;
+  const { newPassword, confirmation } = req.body;
+
+  const user = await User.findOne({ where: { resetToken } });
+
+  if (!user) {
+    throw ApiError.badRequest('User not found');
+  }
+
+  if (!resetToken) {
```

Comment:
The check for `resetToken` should occur before querying the database for the user. This ensures that invalid tokens are caught early, potentially reducing unnecessary database queries and improving security.